api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -85,6 +85,18 @@ type OpenStackDataPlaneServiceSpec struct {
 	// CACerts - Secret containing the CA certificate chain
 	// +kubebuilder:validation:Optional
 	CACerts string `json:"caCerts,omitempty" yaml:"caCerts,omitempty"`
+
+	// DeployOnAllNodesets - should the service be deploy across all nodesets
+	// This will override default target of a service play, setting it to 'all'.
+	// If both DeoployOnAllNodesets and PlayTargetOverride are set, default to 'all'.
+	// +kubebuilder:validation:Optional
+	DeployOnAllNodesets bool `json:"deployOnAllNodesets,omitempty" yaml:"deployOnAllNodesets,omitempty"`
+
+	// PlayTargetOverride - alternative target for plays
+	// Target host of the play will be overriden with provided string.
+	// If both DeoployOnAllNodesets and PlayTargetOverride are set, default to 'all'.
+	// +kubebuilder:validation:Optional
+	PlayTargetOverride string `json:"playTargetOverride,omitempty" yaml:"playTargetOverride,omitempty"`
```

Comment:
I don't see why this is needed. Why add a field we'd need to support when we don't really have a use case for it?

Reply:
Playbooks we have in edpm-ansible now are targeting all hosts by default. We are going to allow overrides with the https://github.com/openstack-k8s-operators/edpm-ansible/pull/548 . But the only override we can provide would be, again `all`, to deploy everywhere.

That is fine in the narrow sense of the epic. We could stick with that and call it a day. But what if we wanted to deploy some service on nodes that were not in the original playbook target, but not on `all` either?

To me it seems like relatively likely scenario, and since we are already implementing most of this anyway with the `all` override, I figured we may want to anticipate the need.